crates/ruma-state-res/src/room_version.rs

```diff
@@ -72,6 +76,12 @@ impl RoomVersion {
             RoomVersionId::Version4 => Self::version_4(),
             RoomVersionId::Version5 => Self::version_5(),
             RoomVersionId::Version6 => Self::version_6(),
+            #[cfg(feature = "unstable-pre-spec")]
+            RoomVersionId::Version7 => Self::version_7(),
```

Comment:
For this to work reliably, state-res' `unstable-pre-spec` feature has to activate identifiers' pre-spec feature.